libraries/WiFi/src/WiFiServer.cpp

```diff
@@ -103,8 +104,16 @@ bool WiFiServer::hasClient() {
 }
 
 void WiFiServer::end(){
-  close(sockfd);
+  lwip_close_r(sockfd);
   sockfd = -1;
   _listening = false;
 }
 
+void WiFiServer::close(){
```

Comment:
ESP32 should actually always use stop() as that is the official Arduino API. close and end should e deprecated.